lms/djangoapps/courseware/access_response.py

```diff
@@ -0,0 +1,85 @@
+"""
+This file contains all the classes used by has_access for error handling
+"""
+
+
+class AccessResponse(object):
+    """Class that represents a response from a has_access permission check."""
+    def __init__(self, has_access, error_code=None, developer_message=None, user_message=None):
+        """
+        :param has_access (bool): if the user is granted access or not
+        :param error_code (String): optional - default is None. unique identifier for the specific type of error
+        :param developer_message (String): optional - default is None. message to show the developer
+        :param user_message (String): optional - default is None. message to show the user
+        """
+        self.has_access = has_access
+        self.error_code = error_code
+        self.developer_message = developer_message
+        self.user_message = user_message
+        if has_access:
+            assert error_code is None
+
+    def __nonzero__(self):
+        """Overrides bool() to correct truth value testing"""
```

Comment:
I would expand on this docstring.  It's not just for test code.  It makes it easier for callers that just want to verify the truth value of the accessibility and not forward the error.